control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -185,10 +185,62 @@ impl ResourceResize for OperationGuardArc<VolumeSpec> {
 
     async fn resize(
         &mut self,
-        _registry: &Registry,
-        _request: &Self::Resize,
+        registry: &Registry,
+        request: &Self::Resize,
     ) -> Result<Self::ResizeOutput, SvcError> {
-        unimplemented!()
+        let specs = registry.specs();
+        let spec = self.as_ref().clone();
+        let state = registry.volume_state(&request.uuid).await?;
+
+        // Pre-Check - volume state ok for resize.
+        if !spec.vol_status_ok_for_resize() {
+            return Err(SvcError::InUse {
+                kind: stor_port::transport_api::ResourceKind::Volume,
+                id: spec.uuid.to_string(),
+            });
+        }
+
+        let spec_clone = self
+            .start_update(
+                registry,
+                &state,
+                VolumeOperation::Resize(request.requested_size),
+            )
+            .await?;
+        // For the volume being resized, resize each replica of the volume. For a failure to
+        // any of the replica, the volume resize operation is to be deemed failure.
+        let replicas = specs.volume_replicas(&request.uuid);
+        for replica in replicas {
+            let mut replica = replica.operation_guard_wait().await?;
```

Comment:
Should we not continue if we cannot take the guard?

Reply:
If we do that, by design the `volume resize` operation is failure as a whole if even one replica is left out - and the added space from replicas that get resized will anyway be potentially reclaimed.